orchestra/project_api/views.py

```diff
@@ -157,10 +164,93 @@ def message_project_team(request):
     return {'success': True}
 
 
+@api_endpoint(methods=['POST'],
+              permissions=(IsSignedUser,),
+              logger=logger,
+              auths=(OrchestraProjectAPIAuthentication,))
+def create_todos_from_template(request):
+    """
+    Endpoint for creating todos in a project.
+    Payload example:
+    {
+        'todolist_template_slug': 'some-template-slug-123',
+        'step_slug': 'some-step-slug-123',
+        'project_id': 'some-project-id-123'
+        'additional_data': {
+            'some_key': 'some_value'
+        }
+    }
+    """
+    data = load_encoded_json(request.body)
+    try:
+        todolist_template_slug = data.get('todolist_template_slug')
+        step_slug = data.get('step_slug')
+        project_id = data.get('project_id')
+        additional_data = data.get('additional_data')
+        if step_slug and project_id and todolist_template_slug:
+            add_todolist_template(todolist_template_slug, project_id,
+                                  step_slug, additional_data)
+            todos = Todo.objects.filter(
+                template__slug=todolist_template_slug,
+                project__id=project_id,
+                step__slug=step_slug).order_by('-created_at')
+            serializer = BulkTodoSerializerWithoutQA(todos, many=True)
+            return {
+                'success': True,
+                'todos': serializer.data
+            }
+        else:
+            text = ('An object with `template_slug`, `step_slug`,'
+                    ' and `project_id` attributes should be supplied')
+            raise Exception(text)
+    except Exception as e:
+        return {
+            'success': False,
+            'errors': {
+                'error': str(e)
+            }
+        }
+
+
+@api_endpoint(methods=['POST'],
+              permissions=(IsSignedUser,),
+              logger=logger,
+              auths=(OrchestraProjectAPIAuthentication,))
+def todo_sections_starting_order(request):
```

Comment:
@jumasheff is doing the code review, so i'll just focus on this one endpoint to make sure I understand its purpose.

I can't think of an experience in Orchestra or the B12 product that requires this order from Orchestra, as the product currently decides on the fixed section order on its end. Since Orchestra doesn't model sections deeply (maybe it should---right now they are just strings), it seems like the metadata for the section order doesn't have a great home in Orchestra without more data modelling.

Because of the lack of model, the implementation here orders sections by count, which is different from the behavior in the product. I'm open to ideas, but I can imagine two worlds
- Make it so sections are just strings in Orchestra. The caller (e.g., B12 product) can determine their display order.
- Push section metadata like section order into Orchestra's data model for to-do lists, and as a result, add that information to the existing to-do endpoints.

Reply:
Got rid of the API.